tests/devices/default_qubit/test_default_qubit_native_mcm.py

```diff
@@ -344,9 +368,14 @@ def func(x, y, z):
     results1 = func1(*params)
     results2 = func2(*params)
 
-    if postselect is None or measure_f in (qml.expval, qml.probs, qml.var):
-        for r1, r2 in zip(results1, results2):
-            validate_measurements(measure_f, shots, r1, r2)
+    if isinstance(shots, Sequence):
+        for s, r1, r2 in zip(shots, results1, results2):
+            for _r1, _r2 in zip(r1, r2):
+                validate_measurements(measure_f, s, _r1, _r2)
```

Comment:
Doesn't `validate_measurements` do this with its recursive structure? :thinking:

Reply:
Due to the changes I made, the handling of validation for multiple measurements became a bit worse. We shouldn't have to iterate over the shot vector, but we do need to iterate over the measurements dim. I can't do fancy indexing with tuples, and can't cast to arrays due to potential ragged dimensions with `qml.sample` as well as type check validation. I will take another look at this tomorrow; I agree it can be cleaner.